### PR TITLE
Create excerpt field option for reporting grid

### PR DESCRIPTION
**Pull request by rachaelob**

### CODE REVIEW

**Comment by beneverard** [comment]
@rachaelob I'd prefer to see the "Content Type" field here type a radio option, it keeps things open for the future and the condition(s) in the `.twig` file will be self-documenting, whereas `{% if card.content_type == true %}` isn't super clear.

File: app/Providers/Blocks/OurReportingGridServiceProvider.php [hunk 1]
```diff
@@ -31,7 +31,9 @@ public function render($block, $content = '', $is_preview = false, $post_id = 0)
         $context = Timber::get_context();
 
         $context['block']['heading'] = get_field('heading');
+        $context['block']['content_type'] = get_field('content_type');
```
Comment (beneverard) [inline app/Providers/Blocks/OurReportingGridServiceProvider.php:34]: Without a default value set no content is displayed:

CleanShot 2024-04-05 at 16 58 38

You'll need something like `get_field('content_type') ?? 'card-strapline'` to ensure this value is used as a fallback.

File: resources/scss/modular/_our-reporting-grid.scss [hunk 2]
```diff
@@ -106,6 +106,10 @@ $font-size: 80px;
             }
         }
 
+        .our-reporting-card__excerpt {
+            font-weight: 300;
+        }
+
```
Comment (beneverard) [inline resources/scss/modular/_our-reporting-grid.scss:112]: I would consider doing the opposite here, making the card content `font-weight: 300`, and then explicitly setting the excerpt to 300. My reasoning here is that in the future if we make the excerpt a WYSIWYG, bold text will likely not work correctly.

File: views/blocks/our-reporting-grid.twig [hunk 3]
```diff
@@ -25,7 +25,11 @@
             >
                 <div class="our-reporting-card__main">
                     <span class="our-reporting-card__heading">{{ card.heading }}</span>
-                    <p class="our-reporting-card__strapline">{{ card.strapline }}</p>
+                    {% if card.content_type == strapline %}
+                        <p class="our-reporting-card__strapline">{{ card.strapline }}</p>
+                    {% elseif card.content_type == excerpt %}
+                        <p class="our-reporting-card__excerpt">{{card.excerpt}}</p>
+                    {% endif %}
```
Comment (beneverard) [inline views/blocks/our-reporting-grid.twig:32]: Two points here, can you make sure these values are strings, but also I don't think the values match up with the values in ACF? They're meant to be `card_strapline` etc?

File: views/blocks/our-reporting-grid.twig [hunk 4]
```diff
@@ -25,7 +25,11 @@
             >
                 <div class="our-reporting-card__main">
                     <span class="our-reporting-card__heading">{{ card.heading }}</span>
```
Comment (beneverard) [inline views/blocks/our-reporting-grid.twig:27]: Now we have the condition below, can we add a new line to separate the simple card heading line with the complex (in that, it has children) if/else statement

**Review by beneverard — CHANGES_REQUESTED** [review]
@rachaelob Some code quality improvements, and some bugs too.

File: app/Providers/Blocks/OurReportingGridServiceProvider.php (hunk 1 above)
Comment (rachaelob) [reply app/Providers/Blocks/OurReportingGridServiceProvider.php:34]: Ahh got it, sorry I missed that during the switch from a true/false to radio buttons

File: acf-json/group_6469c5e35b44c.json [hunk 5]
```diff
@@ -89,21 +89,92 @@
...
+                    "other_choice": 0,
+                    "layout": "vertical",
+                    "save_other_choice": 0,
+                    "parent_repeater": "field_6469c5e3696cf"
+                },
                 {
                     "key": "field_6469c5e3f045f",
                     "label": "Strapline",
-                    "name": "strapline",
+                    "name": "card_strapline",
```
Comment (beneverard) [inline acf-json/group_6469c5e35b44c.json:124]: @rachaelob Sorry I hadn't noticed this before, the strapline field name must not be changed, doing so means any existing strapline data cannot be used. In addition, it doesn't need to be namespaced (the `card_` bit), as it's clear what this belongs to.

Can you remove `card_` from both fields, and update any references to this also.

File: resources/scss/modular/_our-reporting-grid.scss [hunk 6]
```diff
@@ -83,6 +83,10 @@ $font-size: 80px;
 
     .our-reporting-card__main {
         margin-bottom: auto;
+        font-weight: 400;
+    }
+    .our-reporting-card__main:last-child {
```
Comment (beneverard) [inline resources/scss/modular/_our-reporting-grid.scss:88]: @rachaelob Extra space above needed to separate these

**Review by beneverard — CHANGES_REQUESTED** [review]
@rachaelob An issue here that I didn't pick up before, the strapline field name must not be changed, as it means any existing content will not work.

**Comment by beneverard** [comment]
@rachaelob All good, I've merged into develop, you can delete your local feature branch now and start your release.

**Pull request merged**